terraform/environments/data-platform/glue_job.tf

```diff
@@ -0,0 +1,122 @@
+
+locals {
+  default_arguments = {
+    "--continuous-log-logGroup" = aws_cloudwatch_log_group.log_group.name
+  }
+}
+
+
+# load the glue script to bucket as bucket object
+resource "aws_s3_object" "object" {
+  bucket=module.s3-bucket.bucket.id
+  key    = "glue_script/glue_transform_script.py"
+  source = "glue_script/glue_transform_script.py"
+  etag = filemd5("glue_script/glue_transform_script.py")
+}
+
+resource "aws_glue_job" "glue_job" {
+  name        = "${local.name}-glue-job-${local.environment}"
+  role_arn    = aws_iam_role.glue-service-role.arn
+  description            = "Glue job to transform and load data products to the glue catalog"
+  glue_version           = local.glue_version
+  max_retries            = local.max_retries
+  security_configuration = aws_glue_security_configuration.sec_cfg.id
+  worker_type            = local.worker_type
+  number_of_workers      = local.number_of_workers
+  timeout                = local.timeout
+  execution_class        = local.execution_class
+  tags                   = local.tags
+
+  command {
+    script_location = "s3://${module.s3-bucket.bucket.id}/glue_script/glue_transform_script.py"
+    name            = "glueetl"
+  }
+
+  default_arguments = merge(local.glue_default_arguments, local.default_arguments)
+
+  execution_property {
+    max_concurrent_runs = local.max_concurrent
+  }
+}
+
+### Glue Job Service Role
+resource "aws_iam_role" "glue-service-role" {
+  name  = "${local.name}-glue-role-${local.environment}"
+  tags  = local.tags
+  path  = "/"
+
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "sts:AssumeRole",
+            "Principal": {
+               "Service": "glue.amazonaws.com"
+            },
+            "Effect": "Allow",
+            "Sid": ""
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_policy_attachment" "glue-service-policy" {
+  name       = "${local.name}-role-attach-${local.environment}"
+  roles      = [aws_iam_role.glue-service-role.name]
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
+}
+
+data "aws_iam_policy_document" "s3-access" {
+  statement {
+    sid = "GetPutAccess"
```

Comment:
Can the policy be the format below rather than json policy ? 

data "aws_iam_policy_document" "iam_policy_document_for_lambda" {
  statement {
    sid       = "GETPUTBucketAccess"
    effect    = "Allow"
    actions   = ["s3:GetObject*", "s3:PutObject*"]
    resources = ["${module.s3-bucket.bucket.arn}/code/*"]
  }
}

Reply:
Think this one already is?